airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -207,4 +262,41 @@ private Map<String, Object> generateDestinationMetadata(final UUID destinationDe
     return TrackingMetadata.generateDestinationDefinitionMetadata(destinationDefinition);
   }
 
+  @VisibleForTesting
+  List<String> buildJsonPathFromOAuthFlowInitParameters(final List<List<String>> oAuthFlowInitParameters) {
+    return oAuthFlowInitParameters.stream()
+        .map(path -> "$." + String.join(".", path))
+        .toList();
+  }
+
+  @VisibleForTesting
+  JsonNode getOauthFromDBIfNeeded(final JsonNode oAuthInputConfigurationFromDB, final JsonNode oAuthInputConfigurationFromInput) {
+    final Map<String, String> result = new HashMap<>();
+
+    Jsons.deserializeToStringMap(oAuthInputConfigurationFromInput)
+        .forEach((k, v) -> {
+          if (AirbyteSecretConstants.SECRETS_MASK.equals(v)) {
+            if (oAuthInputConfigurationFromDB.has(k)) {
+              result.put(k, oAuthInputConfigurationFromDB.get(k).textValue());
+            } else {
+              LOGGER.warn("Missing the k {} in the config store in DB", k);
+            }
+
+          } else {
+            result.put(k, v);
+          }
+        });
+
+    return Jsons.jsonNode(result);
+  }
+
+  @VisibleForTesting
+  JsonNode getOAuthInputConfiguration(final JsonNode hydratedSourceConnectionConfiguration, final List<String> pathsToGet) {
```

Comment:
From the prototype of this function, looks like we could run into conflicts in the `Collectors.toMap`.
Is that a valid concern? Should we have some type of error handling here?

Reply:
Done,